cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,215 @@
+import java.sql.DriverManager;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.sql.DataSource;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+
+public class Main implements HttpFunction {
+
+  private static final String PROJECT_ID = System.getenv("PROJECT_ID");
+  private static final String CONNECTION_NAME = System.getenv("CLOUD_SQL_CONNECTION_NAME");
+  private static final String DB_NAME = System.getenv("DB_NAME");
+  private static final String DB_USER = System.getenv("DB_USER");
+  private static final String DB_PASSWORD = System.getenv("DB_PASS");
+  private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+
+  private static class LazyInitializer {
+    private static final Tracer TRACER = Tracing.getTracer();
+    static {
+      try {
+        // Set up StackDriver
+        StackdriverTraceExporter.createAndRegister(
+            StackdriverTraceConfiguration.builder()
+                .setProjectId(PROJECT_ID)
+                .build());
+
+      } catch (java.io.IOException e) {
+        Main.LOGGER.log(Level.WARNING, "Warning: Could not set up Stackdriver Trace", e);
+      }
+
+      // Trace every request
+      TraceConfig traceConfig = Tracing.getTraceConfig();
+      traceConfig.updateActiveTraceParams(
+          traceConfig.getActiveTraceParams().toBuilder().setSampler(Samplers.alwaysSample())
+              .build());
+    }
+
+    private LazyInitializer() {
+    }
+
+    private static Tracer getTracer() {
+      return LazyInitializer.TRACER;
+    }
+  }
+
+
+  @Override
+  public void service(HttpRequest request, HttpResponse response)
+      throws IOException {
+
+    Integer duration = Integer.parseInt(request.getFirstQueryParameter("duration").orElse("5000"));
+    Integer interval = Integer.parseInt(request.getFirstQueryParameter("interval").orElse("500"));
+    String connType = request.getFirstQueryParameter("conn_type").orElse("pool");
+
+    String requestId = UUID.randomUUID().toString();
+
+    // Set up URL parameters
+    String jdbcURL = String.format("jdbc:postgresql:///%s", DB_NAME);
+    Properties connProps = new Properties();
+    connProps.setProperty("user", DB_USER);
+    connProps.setProperty("password", DB_PASSWORD);
+    connProps.setProperty("socketFactory", "com.google.cloud.sql.postgres.SocketFactory");
+    connProps.setProperty("cloudSqlInstance", CONNECTION_NAME);
+
+    LOGGER.log(Level.INFO, "Started logging.");
+
+    // Create thread pool
+    ScheduledExecutorService execService = Executors.newScheduledThreadPool(2);
+
+    switch (connType) {
+      case "pool":
+        // Start tests for pooled connections
+        DataSource pool = createPool(jdbcURL, connProps);
+        execService.scheduleAtFixedRate(() -> {
```

Comment:
The lambda here is a `Runnable`, so you can actually use this to your advantage to avoid scheduling in two different places:

```java
Runnable connectFunc;
switch(connType){
  case "pool": 
    connectFunc = () -> connectWithPool(createPool(jdbcURL, connProps), requestId);
    break;
  case "regular":
     connectFunc = () -> connectRegular(jdbcURL, connProps, requestId);
     break;
}
execService.scheduleAtFixedRate(connectFunc, 0, interval, TimeUnit.MILLISECONDS);
```

Reply:
done